tests/v2/validation/public_api/crdgeneration/crd_generation.go

```diff
@@ -0,0 +1,137 @@
+package crdgeneration
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/customresourcedefinitions"
+	"github.com/rancher/rancher/tests/framework/extensions/kubectl"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	groupManagement   = "management.cattle.io"
+	groupCatalog      = "catalog.cattle.io"
+	groupUI           = "ui.cattle.io"
+	groupCluster      = "cluster.cattle.io"
+	groupProject      = "project.cattle.io"
+	groupRKE          = "rke.cattle.io"
+	groupProvisioning = "provisioning.cattle.io"
+	groupFleet        = "fleet.cattle.io"
+	roleTemplate      = "roletemplates.management.cattle.io" //Remove in 2.8.0
+	fleetlocal        = "fleet-local"
+	fleetdefault      = "fleet-default"
+)
+
+func mapCRD(crdList []string) map[string][]string {
+	groups := []string{groupManagement, groupCatalog, groupUI, groupCluster, groupProject, groupRKE, groupProvisioning, groupFleet}
+	crdMap := make(map[string][]string)
+
+	for _, group := range groups {
+		crdMap[group] = []string{}
+	}
+
+	crdMap["additional"] = []string{}
+
+	for _, crd := range crdList {
+		found := false
+		for _, group := range groups {
+			if strings.Contains(crd, group) {
+				crdMap[group] = append(crdMap[group], crd)
+				found = true
+				break
+			}
+		}
+		if !found {
+			crdMap["additional"] = append(crdMap["additional"], crd)
+		}
+	}
+
+	return crdMap
+}
+
+func listCRDS(client *rancher.Client, clusterID string) ([]string, error) {
+	crdCollection, err := customresourcedefinitions.ListCustomResourceDefinitions(client, clusterID, "")
+	if err != nil {
+		return nil, err
+	}
+
+	crds := make([]string, len(crdCollection.Items))
+	for idx, crd := range crdCollection.Items {
+		crds[idx] = crd.GetName()
+	}
+	return crds, nil
+}
+
+func validateCRDList(t *testing.T, crdsList []string, crdMapPreUpgrade map[string][]string, clusterName string) {
+	crdMap := mapCRD(crdsList)
+
+	if clusterName == "local" {
```

Comment:
should we add a 'warn' log or something if the cluster name is not local, so that we know no group assertions were checked?

Reply:
For the current release aka 2.7.7, we are only generating the crds for local cluster and these crds are only present in the local cluster. They not available in the downstream cluster. So, adding a warn log would not be appropriate here and also log message is not needed - updated rb.Run() logs the clusterName here: https://github.com/rancher/rancher/blob/8607ca927ecd4c55dc18ac36ec80d37ef8fd165e/tests/v2/validation/public_api/crdgeneration/crd_generation_test.go#L77